stdlib/public/Concurrency/TaskGroup.swift

```diff
@@ -104,60 +97,54 @@ public func withTaskGroup<ChildTaskResult: Sendable, GroupResult>(
-/// ### Implicit awaiting
-/// When the group returns it will implicitly await for all spawned tasks to
-/// complete. The tasks are only cancelled if `cancelAll()` was invoked before
-/// returning, the groups' task was cancelled, or the group body has thrown.
+/// When the group returns,
+/// it implicitly waits for all spawned tasks to complete.
+/// The tasks are canceled only if `cancelAll()` was invoked before returning,
+/// if the groups' task was canceled,
+/// or if the group body throws an error.
 ///
-/// When results of tasks added to the group need to be collected, one can
-/// gather their results using the following pattern:
+/// To collect the results of tasks that were added to the group,
+/// use the following pattern:
 ///
-///     while let result = await try group.next() {
-///       // some accumulation logic (e.g. sum += result)
+///     while let result = await group.next() {
+///         // some accumulation logic (e.g. sum += result)
 ///     }
 ///
 /// It is also possible to collect results from the group by using its
 /// `AsyncSequence` conformance, which enables its use in an asynchronous for-loop,
 /// like this:
 ///
-///     for try await result in group {
-///       // some accumulation logic (e.g. sum += result)
+///     for await result in group {
+///         // some accumulation logic (e.g. sum += result)
 ///      }
+/// ◊TR: Which of the above do we prefer?  Why give two recommendations?
+///
+/// When tasks are added to the group
+/// using the `Group.spawn(priority:operation:)` method,
+/// they might begin executing immediately.
+/// Even if their results are not collected explicitly and such task throws,
+/// and was not yet canceled,
+/// it may result in the `withTaskGroup` throwing.
+/// ◊TR: What does the above "such task throws" mean?
 ///
-/// ### Thrown errors
-/// When tasks are added to the group using the `group.spawn` function, they may
-/// immediately begin executing. Even if their results are not collected explicitly
-/// and such task throws, and was not yet cancelled, it may result in the `withTaskGroup`
-/// throwing.
+/// Task Group Cancellation
+/// =======================
 ///
-/// ### Cancellation
-/// If the task that the group is running in is cancelled, the group becomes 
-/// cancelled and all child tasks spawned in the group are cancelled as well.
-/// 
-/// If an error is thrown out of the task group, all of its remaining tasks
-/// will be cancelled and the `withTaskGroup` call will rethrow that error.
+/// Canceling the task in which the group is running
+/// also cancels the group and all of its child tasks.
 ///
-/// Individual tasks throwing results in their corresponding `try group.next()`
-/// call throwing, giving a chance to handle individual errors or letting the
-/// error be rethrown by the group.
+/// If an error is thrown by one of the tasks in a task group,
+/// all of its remaining tasks are canceled,
+/// and the `withTaskGroup` method rethrows that error.
```

Comment:
That simplification sadly is wrong, we must talk about the `try group.next()` throwing.

In this case:

```swift
{ group in 
group.spawn { throw Boom() } 
}
```

nothing is cancelled, while in this case all other tasks _are_ cancelled:

```swift
{ group in 
group.spawn { throw Boom() } 
try group.next()! // throws boom, exits the group, causes cancelAll
}
```